packages/components/src/components/tabs/tabs.tsx

```diff
@@ -116,6 +118,17 @@ const Tabs = ({
             onTabItemClick?.(active_tab_index);
         }
         setActiveLineStyle();
+        if (should_scroll_tab_into_view) {
+            setTimeout(
```

Comment:
We need to wrap our code with `setTimeout` for proper autoscrolling during the initial (first) render. 

> The `scrollIntoView` method might not work as expected when called directly inside the `useEffect` hook because of the timing of when it's executed.
> In React, the `useEffect` hook runs after the initial render and subsequent updates. However, when `scrollIntoView` is called directly inside the `useEffect`, it might not behave as expected because the DOM may not have been fully updated or the elements we are trying to scroll to may not have been added to the DOM at the time the `useEffect` runs.

Reply:
@heorhi-deriv we have a new feature on the css which is   `scroll-snap-type`, can you try this instead of doing that with js?
```suggestion
            .container {
                  overflow-x: scroll;
                   scroll-snap-type: x mandatory;
                   &.item{
                      scroll-snap-align: center;
                   }
            }
```

Reply:
That `setTimeout` is worrying me. What if the component unmounts between the callback being executed, and the setTimeout calling it again.

I guess it would be rare right. Should we clean those timers up on unmount?

```ts
        const timeoutId = setTimeout(() => {
            setActiveLineStyle();
        }, 500);
        setActiveTimeoutId(timeoutId);
    }
}, [activeTimeoutId]);

// Cleanup on component unmount
React.useEffect(() => {
    return () => {
        if (activeTimeoutId) {
            clearTimeout(activeTimeoutId);
        }
    };
}, [activeTimeoutId]);
```

Bit horrible though.

Reply:
Or what about something like this?

```ts
    const setActiveLineStyle = useCallback(() => {
        const tabs_wrapper_bounds = tabs_wrapper_ref?.current?.getBoundingClientRect();
        const active_tab_bounds = active_tab_ref?.current?.getBoundingClientRect();

        if (tabs_wrapper_bounds && active_tab_bounds) {
            updateActiveLineStyle({
                left: active_tab_bounds.left - tabs_wrapper_bounds.left,
                width: active_tab_bounds.width,
            });
        }
    }, []);

    useEffect(() => {
        let timeoutId: NodeJS.Timeout;

        if (!active_tab_ref.current || !tabs_wrapper_ref.current) {
            timeoutId = setTimeout(setActiveLineStyle, 500);
        } else {
            setActiveLineStyle();
        }

        return () => {
            clearTimeout(timeoutId);
        };
    }, [setActiveLineStyle]);
```

Reply:
hey hey @markwylde-deriv 
good catch 👍
let me add a cleanup for setTimeout

Reply:
hey hey @mahdiyeh-deriv 
i checked your suggestion and would like to summarise my investigations:
#### Approach with `scroll-snap-type: x mandatory`
1. Not critical but still, switching between the tabs are not smoothly.
2. Critical thing is that when i try to scroll tabs, there is a case that i can't scroll to `Deposit` tab when my active tab is `Withdraw`, because it always tries me to return to `Withdraw` tab (pls check the video).
#### Approach with `setTimeout`
1. Smoothly switching between the tabs
2. Can scroll the tabs without sticking to particular tab (pls check the video).

`scroll-snap-type: x mandatory`
https://github.com/binary-com/deriv-app/assets/103181646/4541d881-1aac-473e-a8ef-97dc9061981f

`setTimeout`
https://github.com/binary-com/deriv-app/assets/103181646/80be7d58-0ba5-4c0f-bf44-ea6185828dfd

Reply:
@heorhi-deriv  So based on the recordings let's keep the setTimeout approach 👍